app/frontend/src/features/userQueries/useUserByUsername.test.tsx

```diff
@@ -0,0 +1,199 @@
+      error: "",
+      data: undefined,
+    });
+  });
+});
+
+describe("when user has loaded", () => {
+  it("returns the user data with no errors", async () => {
+    const { result, waitFor } = renderHook(
+      () => useUserByUsername("funnydog"),
+      {
+        wrapper,
+      }
+    );
+    await waitFor(() => !result.current.isLoading);
+
+    expect(result.current).toEqual({
+      isLoading: false,
+      isFetching: false,
+      isError: false,
+      error: "",
+      data: {
+        name: "Funny Dog",
+        userId: 2,
+        username: "funnydog",
+        avatarUrl: "funnydog.jpg",
+      },
+    });
+    expect(getUserMock).toHaveBeenCalledTimes(2);
+  });
+
+  it("returns isError as true and the errors if some getUser queries failed", async () => {
+    getUserMock.mockRejectedValue(new Error("Error fetching user funnydog"));
+
+    const { result, waitForNextUpdate } = renderHook(
+      () => useUserByUsername("funnydog"),
+      {
+        wrapper,
+      }
+    );
+    await waitForNextUpdate();
+
+    expect(result.current).toMatchObject({
+      isLoading: false,
+      isFetching: false,
+      isError: true,
+      error: "Error fetching user funnydog",
+      data: undefined,
+    });
+  });
+});
+
+describe("cached data", () => {
+  it("is used instead of refetching", async () => {
+    const sharedClient = new QueryClient({
+      defaultOptions: { queries: { retry: false } },
+    });
+    const sharedClientWrapper = ({
+      children,
+    }: {
+      children: React.ReactNode;
+    }) => (
+      <QueryClientProvider client={sharedClient}>
+        {children}
+      </QueryClientProvider>
+    );
+    const { waitForNextUpdate } = renderHook(
+      () => useUserByUsername("funnydog"),
+      {
+        wrapper: sharedClientWrapper,
+      }
+    );
+    await waitForNextUpdate();
+    expect(getUserMock).toBeCalledTimes(2);
+
+    const { result } = renderHook(() => useUserByUsername("funnydog"), {
+      wrapper: sharedClientWrapper,
+    });
+    expect(getUserMock).toBeCalledTimes(2);
+    expect(result.current.isFetching).toBe(false);
+    expect(result.current.isLoading).toBe(false);
+  });
+
+  it("is invalidated when requested, userid2user map not invalidated", async () => {
+    const sharedClient = new QueryClient({
+      defaultOptions: { queries: { retry: false } },
+    });
+    const sharedClientWrapper = ({
+      children,
+    }: {
+      children: React.ReactNode;
+    }) => (
+      <QueryClientProvider client={sharedClient}>
+        {children}
+      </QueryClientProvider>
+    );
+    const { waitForNextUpdate } = renderHook(
+      () => useUserByUsername("funnydog"),
+      {
+        wrapper: sharedClientWrapper,
+      }
+    );
+    await waitForNextUpdate();
+    expect(getUserMock).toBeCalledTimes(2);
+
+    renderHook(() => useUserByUsername("funnydog", true), {
+      wrapper: sharedClientWrapper,
+    });
+
+    expect(getUserMock).toBeCalledTimes(3);
```

Comment:
Following the above point, we'd be able to see `getUserMock` gets called exactly once here due to the invalidation rather than having to remember it's invalidation call + 2 caused by the data seeding that we didn't care about.